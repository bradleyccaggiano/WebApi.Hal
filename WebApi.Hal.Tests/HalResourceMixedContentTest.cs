using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Http;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using WebApi.Hal.Tests.Representations;
using Xunit;

namespace WebApi.Hal.Tests
{
    public class HalResourceMixedContentTest
    {
        readonly OrganisationWithPeopleDetailRepresentation resource;

        public HalResourceMixedContentTest()
        {
            resource = new OrganisationWithPeopleDetailRepresentation(1, "Org Name")
            {
                Boss = new Boss(2, "Eunice PHB", 1, true)
            };
            resource.People.Add(new Person(3, "Dilbert", 1));
            resource.People.Add(new Person(4, "Wally", 1));
            resource.People.Add(new Person(5, "Alice", 1));
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void peopledetail_get_json_test()
        {
            // arrange
            var mediaFormatter = new JsonHalMediaTypeFormatter { Indent = true };
            var content = new StringContent(string.Empty);
            var type = resource.GetType();

            // act
            using (var stream = new MemoryStream())
            {
                mediaFormatter.WriteToStreamAsync(type, resource, stream, content, null).Wait();
                stream.Seek(0, SeekOrigin.Begin);
                var serialisedResult = new StreamReader(stream).ReadToEnd();

                // assert
                Approvals.Verify(serialisedResult);
            }
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void peopledetail_get_xml_test()
        {
            // arrange
            var mediaFormatter = new XmlHalMediaTypeFormatter();
            var content = new StringContent(string.Empty);
            var type = resource.GetType();

            // act
            using (var stream = new MemoryStream())
            {
                mediaFormatter.WriteToStreamAsync(type, resource, stream, content, null);
                stream.Seek(0, SeekOrigin.Begin);
                var serialisedResult = new StreamReader(stream).ReadToEnd();

                // assert
                Approvals.Verify(serialisedResult);
            }
        } 
    }
}
