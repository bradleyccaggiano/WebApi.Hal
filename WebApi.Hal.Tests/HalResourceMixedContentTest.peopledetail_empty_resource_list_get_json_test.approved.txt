{
  "Id": 1,
  "Name": "Org Name",
  "_links": {
    "self": {
      "href": "/api/organisations/1"
    },
    "people": {
      "href": "/api/organisations/1/people"
    },
    "boss": {
      "href": "/api/organisations/1/boss"
    }
  },
  "_embedded": {
    "person": [],
    "boss": {
      "HasPointyHair": true,
      "Id": 2,
      "Name": "Eunice PHB",
      "OrganisationId": 1,
      "_links": {
        "self": {
          "href": "/api/organisations/1/boss"
        }
      }
    }
  }
}